### ISSUE TITLE
Syntax highlighter is incomplete and needs clearer error messages

### ISSUE THREAD

**Issue by Hades32** [KronicDeth/intellij-elixir#6]
Even a simple file like this is completely red and no sensible Syntax highlighting.
```
defmodule Chat do
  use Application

  # See http://elixir-lang.org/docs/stable/Application.Behaviour.html
  # for more information on OTP Applications
  def start(_type, _args) do
    Chat.Supervisor.start_link
  end
end
```
The error is something like "ElixirTokenType... or ...[..] expected, but got 'd' "

**Comment by elfet** [comment]
@KronicDeth are you going to release this plugin? %)

**Comment by KronicDeth** [comment]
@elfet some version of the plugin has been released in the JetBrains plugin repository since 2014-04-08.  You can install it using [the instructions in the README](https://github.com/KronicDeth/intellij-elixir#installation).  This issue is still open because the parser won't be complete until [v1.0.0](https://github.com/KronicDeth/intellij-elixir/milestones/v1.0.0).

**Comment by nakhli** [comment]
off topic: do you guys accept donations for the project? it would be nice if we are able to accelerate this work.

**Comment by smeevil** [comment]
+1 on funding, really would like to have first class support in ruby mine(or derivative) just like alchemist dies for emacs

**Comment by nakhli** [comment]
+1 for RubyMine! I use rubymine for both ruby and golang

**Comment by KronicDeth** [comment]
@smeevil, @nakhli the plugin does work in Rubymine.  You can make `*.ex` or `*.exs` files and they'll be lexed and parsed and annotated as Elixir.

The only feature that doesn't is "File > New > Project From Existing Source..." because that [uses](https://github.com/KronicDeth/intellij-elixir/blob/10e0951e2545c8f75d636f8a29bc8a34484a511d/src/META-INF/rich-platform-plugin.xml#L7) an Extension Point (projectStructureDetector) that JetBrains only supplies for IntelliJ and not for the language-specific IDEs like Rubymine.  Extension Points that only work in IntelliJ are in https://github.com/JetBrains/intellij-community/blob/master/resources/src/idea/RichPlatformPlugin.xml.  There is probably a similar extension point in non-Intellij IDEs, since Rubymine does support projects, but it's not like IntelliJ projects where you can select different langauges is my understanding since the Rubymine projects involve selecting bootstrap or a different JS framework to use in a Ruby project like Angular or Ember.

@nakhli are you saying that golang plugin allows you to create Go-only projects?  If it's open-source I can look at what it's doing.

**Comment by smeevil** [comment]
yeah , I am using it in ruby mine :) I was referring to the support and development of the plugin in general. So if you like feel free to put a donation button somewhere !

**Comment by KronicDeth** [comment]
@smeevil, @nakhli I added a Paypal Donate link to the [README](https://github.com/KronicDeth/intellij-elixir/blob/master/README.md#donations).  I can look into other payment systems (Patreon, etc) if that doesn't work with however you wanted to donate.

**Comment by nakhli** [comment]
@KronicDeth I am using this go plugin: https://github.com/go-lang-plugin-org

**Comment by olivermt** [comment]
Just threw up a $100 donation to you, good work so far :+1: 

I was wondering if you have some sort of timeline for reaching 1.0?

**Comment by smeevil** [comment]
Added an other $100 on top of that as well, keep it up :D

**Comment by KronicDeth** [comment]
@olivermt 1.0 is a [milestone](https://github.com/KronicDeth/intellij-elixir/milestones/v1.0.0), so you can just monitor that to see what's blocking.  Right now, the only thing I know is the [bug in Kernel.ex](https://github.com/KronicDeth/intellij-elixir/issues/172).  I'll also probably make a feature to ensure all the tokens and rules have human-readable names for clearer error messages since that's part of this issue in  @develop7's [screenshot](https://github.com/KronicDeth/intellij-elixir/issues/6#issuecomment-59082494)

**Comment by KronicDeth** [comment]
#172 was fixed by #173.  Better human-readable error messages by way of token names landed in #178.  Expect the v1.0.0 release sometime this week.  Depends on how many changes I need to make to the README.  (I keep changes in the CHANGELOG.md and pluginx.ml, but out of the README on master so that users don't open bugs asking where a feature in the README is in the released version that lags master.)

**Comment by develop7** [comment]
@KronicDeth thanks!
